corehq/apps/userreports/sql/adapter.py

```diff
@@ -123,6 +123,11 @@ def delete(self, doc):
             delete = table.delete(table.c.doc_id == doc['_id'])
             connection.execute(delete)
 
+    def doc_exists(self, doc):
+        return self.session_helper.Session.query(
+            self.get_query_object().filter_by(doc_id=doc['_id']).exists()
```

Comment:
@snopoke I checked this and ```exists()``` doesn't execute the query this only create sub-query which we must execute.